analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,150 @@
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+#Analyse Pathways statistically correlated with clinical variables
+getVarNames <- function(x)
+{
+	tmp <- strsplit(x, split="-")[[1]]
+	tmp1 <- tmp[1]
+	tmp2 <- tmp[2];
+	return(c(tmp1, tmp2))
+}
+
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,]);
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar")
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95);
+	posthoc <- data.frame(posthoc[[1]]);
+	posthoc[,"F_Value"] <- summary(aovOut)[[1]][[4]][1];
+	posthoc[,"AOV_P"] <- summary(aovOut)[[1]][[5]][1];
+	posthoc[,"Pathway"] <- pathwayName;
+	posthoc[,"ClinVar"] <- clinVarName;
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc);
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc);
+}
+
+##############################
+#Analysis by Disease
+##############################
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology");
+diseaseType <- do.call("rbind", diseaseType)
+
+#Filter to significant entries
+diseaseTypeFilt <- diseaseType[diseaseType[,"AOV_P"]<0.01,]
+diseaseTypeFilt <- diseaseType[diseaseType[,"p.adj"]<0.05,] 
+
+#Now make all comparisons the same direction & write out results
+diseaseTypeFilt[,"DiseaseX"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarX"]), as.character(diseaseTypeFilt[,"VarY"]))
+diseaseTypeFilt[,"DiseaseY"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarY"]), as.character(diseaseTypeFilt[,"VarX"]))
+diseaseTypeFilt <- diseaseTypeFilt[-1:-2]
+diseaseTypeFilt <- diseaseTypeFilt[, c("DiseaseX", "DiseaseY", colnames(diseaseTypeFilt)[1:8])]
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#Now get diseases highly associated with a certain pathway
+diseaseTableTmp <- data.frame(table(diseaseTypeFilt[,c("DiseaseX", "Pathway")]));
+diseaseTableTmp <- diseaseTableTmp[diseaseTableTmp[,"Freq"]>(length(rownames(geneSetExpMat))*.25),]
+diseaseTableTmp <- diseaseTableTmp[order(-diseaseTableTmp[,"Freq"]),]
+
+#Now set up to create heatmaps
+keepPathways <- unique(diseaseTableTmp[,"Pathway"]);
+keepDisease <- unique(diseaseTableTmp[,"DiseaseX"]);
+tmpClinData <- clinData[c("short_histology", "broad_histology")];
+colnames(tmpClinData)[1] <- "Histology" 
+rownames(geneSetExpMat) <- gsub("HALLMARK", "", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- gsub("_", " ", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- trimws(rownames(geneSetExpMat))
+
+#Heatmap across all samples, only significant pathways
+if(length(keepPathways)>10)
+{
+png("plots/HeatmapPathwaysGenes_all.png", width=1080, height=720)
```

Comment:
Can you indent the arguments for pheatmap? This will make it easier to read.

Reply:
That's fine.